### PR TITLE
Add missing parameters to returns statement in ivtools.sdm.fit_desoto_sandia

### LINKED ISSUE #2315: Parameters missing from ivtools.sdm.fit_desoto_sandia returns statement
**Is your feature request related to a problem? Please describe.**
In addition to the parameters already listed in the returns statement in the docs of [pvlib.ivtools.sdm.fit_desoto_sandia](https://pvlib-python.readthedocs.io/en/stable/_modules/pvlib/ivtools/sdm.html#fit_desoto_sandia), this function also returns `a_ref` and `dEgdT`, which are two parameters required for the [De Soto module performance model](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.pvsystem.calcparams_desoto.html#pvlib-pvsystem-calcparams-desoto).

**Describe the solution you'd like**
Add descriptions of `a_ref` and `dEgdT` to the returns statement in the docs

**Describe alternatives you've considered**
Have I missed something here...?

### PR DESCRIPTION
**Pull request by RDaxini**
- [X] Closes #2315
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing/index.html)
 - [X] Tests added
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [X] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [X] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: pvlib/ivtools/sdm.py [hunk 1]
```diff
@@ -564,25 +564,36 @@ def fit_desoto_sandia(ivcurves, specs, const=None, maxiter=5, eps1=1.e-3):
...
         rsh : array
-            shunt resistance for each IV curve [ohm]
+            Shunt resistance for each IV curve [ohm]
+        a_ref : float
+            The product of the usual diode ideality factor (n, unitless),
+            number of cells in series (Ns), and cell thermal voltage at
+            reference conditions, in units of V.
+        dEgdT : float
+            The temperature dependence of the energy bandgap (Eg) at reference
+            conditions [eV/K]. May be either a scalar value
```
Comment (RDaxini) [inline pvlib/ivtools/sdm.py:590]: These are the correct units, right? 
[ivtools.sdm.fit_desoto_sandia](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.ivtools.sdm.fit_desoto_sandia.html) has this parameter in units of 1/K, but I think that's incorrect.

Comment (kandersolar) [reply pvlib/ivtools/sdm.py:590]: #2154

Comment (RDaxini) [reply pvlib/ivtools/sdm.py]: so [pvlib.ivtools.sdm.fit_desoto](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.ivtools.sdm.fit_desoto.html#pvlib.ivtools.sdm.fit_desoto) is incorrect?

Comment (cwhanse) [reply pvlib/ivtools/sdm.py]: Yes. Unit should be 1/K

Comment (RDaxini) [reply pvlib/ivtools/sdm.py]: #2319

File: pvlib/ivtools/sdm.py [hunk 2]
```diff
@@ -564,25 +564,36 @@ def fit_desoto_sandia(ivcurves, specs, const=None, maxiter=5, eps1=1.e-3):
...
+            The product of the usual diode ideality factor (n, unitless),
+            number of cells in series (Ns), and cell thermal voltage at
+            reference conditions, in units of V.
+        dEgdT : float
+            The temperature dependence of the energy bandgap (Eg) at reference
+            conditions [1/K]. May be either a scalar value
+            (e.g. -0.0002677 as in [1]_) or a DataFrame (this may be useful if
+            dEgdT is a modeled as a function of temperature). For parameters
+            from the SAM CEC module database, dEgdT=-0.0002677 is implicit for
+            all cell types in the parameter estimation algorithm used by NREL.
```
Comment (kandersolar) [inline pvlib/ivtools/sdm.py:594]:
```suggestion
            conditions [1/K].
```

File: pvlib/ivtools/sdm.py [hunk 3]
```diff
@@ -564,25 +564,32 @@ def fit_desoto_sandia(ivcurves, specs, const=None, maxiter=5, eps1=1.e-3):
         I_L_ref : float
             light current at STC [A]
```
Comment (cwhanse) [inline pvlib/ivtools/sdm.py:565]:
```suggestion
            Light current at STC [A]
```

May as well do this line also

**Review by kandersolar — APPROVED** [review]
Thanks @RDaxini

**Pull request merged**